src/validation/validation.cpp

```diff
@@ -2574,7 +2574,6 @@ bool ConnectBlock(const CBlock &block,
 
     // Write undo information to disk
     {
-        WRITELOCK(cs_mapBlockIndex);
```

Comment:
you need the WRITELOCK for `pindex->RaiseValidity(BLOCK_VALID_SCRIPTS);` on line 2595 as this edits the nStatus of the pindex.

Reply:
thanks, fixed.

Reply:
looks good, approved